lib/buffy/throttle_and_timed.ex

```diff
@@ -0,0 +1,372 @@
+
+    - `:supervisor_module` (`atom`) - Optional. A module that implements the `DynamicSupervisor` behaviour. If you are running in a distributed instance, you can set this value to `Horde.DynamicSupervisor`. Defaults to `DynamicSupervisor`.
+
+    - `:supervisor_name` (`atom`) - Optional. The name of the dynamic supervisor to use. Defaults to the built in Buffy dynamic supervisor, but if you are running in a distributed instance you can set this value to a named `Horde.DynamicSupervisor` process. Defaults to `Buffy.DynamicSupervisor`.
+
+    - :throttle (`non_neg_integer`) - Required. The amount of time to wait before invoking the function. This value is in milliseconds.
+
+    - `:loop_interval` (`atom`) - Required. The amount of time that this process will wait while inbox is empty until sending a `:timeout` message (handle via `handle_info`). Resets if message comes in. In milliseconds.
+
+  ## Using with Horde
+
+  If you are running Elixir in a cluster, you can utilize `Horde` to only run one of your throttled functions at a time. To do this, you'll need to set the `:registry_module` and `:supervisor_module` options to `Horde.Registry` and `Horde.DynamicSupervisor` respectively. You'll also need to set the `:registry_name` and `:supervisor_name` options to the name of the Horde registry and dynamic supervisor you want to use.
+
+        defmodule MyThrottler do
+          use Buffy.ThrottleAndTimed,
+            registry_module: Horde.Registry,
+            registry_name: MyApp.HordeRegistry,
+            supervisor_module: Horde.DynamicSupervisor,
+            supervisor_name: MyApp.HordeDynamicSupervisor,
+            throttle: :timer.minutes(2),
+            loop_timeout: :timer.minutes(10)
+
+          def handle_throttle(args) do
+            # Do something with args
+          end
+        end
+
+  ## Telemetry
+
+  These are the events that are called by the `Buffy.ThrottleAndTimed` module:
+
+  - `[:buffy, :throttle, :throttle]` - Emitted when the `throttle/1` function is called.
+  - `[:buffy, :throttle, :timeout]` - Emitted when inbox timeout is triggered.
+  - `[:buffy, :throttle, :handle, :start]` - Emitted at the start of the `handle_throttle/1` function.
+  - `[:buffy, :throttle, :handle, :stop]` - Emitted at the end of the `handle_throttle/1` function.
+  - `[:buffy, :throttle, :handle, :exception]` - Emitted when an error is raised in the `handle_throttle/1` function.
+
+  All of these events will have the following metadata:
+
+  - `:args` - The arguments passed to the `throttle/1` function.
+  - `:key` - A hash of the passed arguments used to deduplicate the throttled function.
+  - `:module` - The module using `Buffy.ThrottleAndTimed`.
+
+  With the additional metadata for `[:buffy, :throttle, :handle, :stop]`:
+
+  - `:result` - The return value of the `handle_throttle/1` function.
+
+  """
+  require Logger
+  alias Buffy.ThrottleAndTimed
```

Comment:
We alias `Buffy.ThrottleAndTimed` here but in certain places we still call `Buffy.ThrottleAndTimed...` and in others we call `ThrottleAndTimed...`. I know this is nitpicky, but do you mind cleaning up the references to `Buffy.ThrottleAndTimed` -> `ThrottleAndTimed` 🙏🏻 😬

Reply:
Done in dc1484e